athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/AwsRestHighLevelClient.java

```diff
@@ -94,6 +98,35 @@ public AwsRestHighLevelClient(RestClientBuilder builder)
         return (LinkedHashMap<String, Object>) mappingsResponse.mappings().get(index).sourceAsMap();
     }
 
+    /**
+     * Retrieves cluster-health information for shards associated with the specified index. The request will time out
+     * if no results are returned after a period of time indicated by timeout.
+     * @param index is used to restrict the request to a specified index.
+     * @param timeout is the command timeout period in seconds.
+     * @return a set of shard ids for the specified index.
+     * @throws IOException if an error occurs while sending the request to the Elasticsearch instance.
+     * @throws RuntimeException if the request times out, or no active-primary shards are present.
+     */
+    public Set<Integer> getShardIds(String index, long timeout)
+            throws RuntimeException, IOException
+    {
+        ClusterHealthRequest request = new ClusterHealthRequest(index)
+                .timeout(new TimeValue(timeout, TimeUnit.SECONDS));
+        // Set request to shard-level details
+        request.level(ClusterHealthRequest.Level.SHARDS);
+
+        ClusterHealthResponse response = cluster().health(request, RequestOptions.DEFAULT);
+
+        if (response.isTimedOut()) {
+            throw new RuntimeException("Request timed out.");
+        }
+        else if (response.getActiveShards() == 0) {
+            throw new RuntimeException("There are no active shards.");
+        }
+
+        return response.getIndices().get(index).getShards().keySet();
```

Comment:
lots of potential for nulls here, isn't there?

Reply:
I'm assuming what you're referring to is the .get(index) which access's a member of a Map. The index is always specified in the request. The behavior that I've seen If the index does not exist in the cluster, is that the request will time-out (throwing a RuntimeException) and won't get to this point. But, just to be on the safe side, i'll include a check that the index is not null and exists in the Map.

Reply:
```java
        else if (index == null || !response.getIndices().containsKey(index)) {
            throw new RuntimeException("Request has an invalid index (" + (index == null ? "null" : index) + ").");
        }
```